keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,593 @@
+     * {@link JsonWebKeyType#EC_HSM EC-HSM}.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Creates a new EC key with P-384 web key curve. The key activates in one day and expires in one year. Subscribes to the call asynchronously
+     * and prints out the newly created ec key details when the callback happens.</p>
+     * <pre>
+     * ECKeyCreateConfig ecKeyCreateConfig = new ECKeyCreateConfig("keyName", JsonWebKeyType.EC)
+     *    .curve(JsonWebKeyCurveName.P_384)
+     *    .notBefore(OffsetDateTime.now().plusDays(1))
+     *    .expires(OffsetDateTime.now().plusYears(1));
+     *
+     * keyAsyncClient.createECKey(ecKeyCreateConfig).subscribe(keyResponse -&gt;
+     *   System.out.printf("EC Key is created with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param ecKeyCreateConfig The key configuration object containing information about the rsa key being created.
+     * @throws NullPointerException if {@code rsaKeyCreateConfig} is {@code null}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    public Mono<Response<Key>> createECKey(ECKeyCreateConfig ecKeyCreateConfig) {
+        KeyRequestParameters parameters = new KeyRequestParameters()
+            .kty(ecKeyCreateConfig.keyType())
+            .curve(ecKeyCreateConfig.curve())
+            .keyOps(ecKeyCreateConfig.keyOperations())
+            .keyAttributes(new KeyRequestAttributes(ecKeyCreateConfig));
+        return service.createKey(endpoint, ecKeyCreateConfig.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Imports an externally created key and stores it in key vault. The import key operation may be used to import any key type
+     * into the Azure Key Vault. If the named key already exists, Azure Key Vault creates a new version of the key. This operation requires the {@code keys/import} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Imports a new key into key vault. Subscribes to the call asynchronously and prints out the newly imported key details
+     * when the callback happens.</p>
+     * <pre>
+     * keyAsyncClient.importKey("keyName", jsonWebKeyToImport).subscribe(keyResponse -&gt;
+     *   System.out.printf("Key is imported with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param name The name for the imported key.
+     * @param keyMaterial The Json web key being imported.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key imported key}.
+     */
+    public Mono<Response<Key>> importKey(String name, JsonWebKey keyMaterial){
```

Comment:
You'll rename `JsonWebKey` and related classes in keys as well?

Reply:
yes, after the arch meeting.

Reply:
We cannot rename JsonWebKey.  It's an implementation of the JOSE standard https://tools.ietf.org/html/rfc7517.

Reply:
@schaabs , but can we rename JsonWebKeyType to KeyType and others similarly,